assets/app/scripts/services/hpa.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("HPAService", function(LimitRangesService) {
+    var getCPURequestToLimitPercent = function(project) {
+      return LimitRangesService.getRequestToLimitPercent('cpu', project);
+    };
+
+    // Converts a percentage of request to a percentage of the limit based on
+    // the request-to-limit ratio.
+    // If CPU request percent is 120% and CPU request-to-limit percent is 50%, returns 60%
+    var convertRequestPercentToLimit = function(requestPercent, project) {
+      if (!requestPercent) {
+        return requestPercent;
+      }
+
+      var cpuRequestToLimitPercent = getCPURequestToLimitPercent(project);
+      var limitPercent = (cpuRequestToLimitPercent / 100) * requestPercent;
+      return Math.round(limitPercent);
+    };
+
+    // Converts a percentage of limit to a percentage of the request based on
+    // the request-to-limit ratio.
+    // If CPU limit percent is 60% and CPU request-to-limit percent is 50%, returns 120%
+    var convertLimitPercentToRequest = function(limitPercent, project) {
+      if (!limitPercent) {
+        return limitPercent;
+      }
+
+      var cpuRequestToLimitPercent = getCPURequestToLimitPercent(project);
+      var requestPercent = limitPercent / (cpuRequestToLimitPercent / 100);
```

Comment:
The function is not called if `cpuRequestToLimitPercent` is 0. Let me add a guard and log a warning anyway.